### ISSUE TITLE
Feature Request: Provide a common subject declaration for register and compatibility tasks

### ISSUE THREAD

**Issue by chriskn** [ImFlog/schema-registry-plugin#116]
Hi, 

I am using this plugin to register and test my AVRO schemas.   
My configuration looks more or less like the one shown in this [example](https://github.com/ImFlog/schema-registry-plugin/blob/master/examples/avro/build.gradle), but has more subjects and references:

```
schemaRegistry {
...
     register {
        subject('company', 'schemas/avro/company.avsc', 'AVRO')
                .addLocalReference("Address", "schemas/avro/location-address.avsc")
        subject('user', 'schemas/avro/user.avsc', 'AVRO')
                .addReference('company', 'company', -1)
        subject('location-address', 'schemas/avro/location-address.avsc', 'AVRO')
        subject('location-latlong', 'schemas/avro/location-latlong.avsc', 'AVRO')
    }
    compatibility {
        subject('company', 'schemas/avro/company_v2.avsc', 'AVRO')
                .addLocalReference("Address", "schemas/avro/location-address.avsc")
        subject('user', 'schemas/avro/user.avsc', 'AVRO')
                .addReference('company', 'company', -1)
        subject('location-address', 'schemas/avro/location-address.avsc', 'AVRO')
        subject('location-latlong', 'schemas/avro/location-latlong.avsc', 'AVRO')
    }
}
```

Due to the duplicate subject definition, I end up with a lot of duplicate configuration. 
Keeping the subjects in sync in the register and compatibility task is error-prone.

To avoid this, I introduced a small abstraction (I only work with local references and AVRO):
 
```
data class Subject(
    val name: String,
    val filePath: String,
    val localReferences: Map<String, String>
) {
    fun toRegisterSubject() = RegisterSubject(
        inputSubject = name,
        file = filePath,
        type = SchemaType.AVRO,
        localReferences = localReferences.map {
            LocalReference(name = it.key, path = it.value)
        }.toMutableList()
    )

    fun toCompatibilitySubject() = CompatibilitySubject(
        inputSubject = name,
        file = filePath,
        type = AVRO,
        localReferences = localReferences.map {
            LocalReference(name = it.key, path = it.value)
        }.toMutableList()
    )
}
```

And configure my register and compatibility task like this:

```
schemaRegistry {
    val subjectA = Subject(...)
    val subjectB = Subject(...)

    compatibility {
        subjects.addAll(
            subjectA.toCompatibilitySubject(),
            subjectB.toCompatibilitySubject()
        )
    }
    register {
        subjects.addAll(
            subjectA.toRegisterSubject(),
            subjectB.toRegisterSubject()
        )
    }
}
```

Would it be possible to add a way to declare subjects outside the specific blocks and reuse them?
Or do I miss something and there is already a way to avoid duplicate configuration?

If help is welcome, let me know. I am willing to contribute.

**Comment by ImFlog** [comment]
Hello @chriskn thank you for using the plugin :)
This is indeed a valid use case !
On the top of my mind I think we could add a method in `CompatibilitySubjectExtension` that would look like :
`fun subject(subject: CompatibilitySubject) = subject.apply { subjects.add(it) }`
Like this in your build.gradle file you can create a `CompatibilitySubject` assign it to a val and reuse it easily.
WDYT ?
If you want to give it a try I will be glad to help, if not this will be my next task.

**Comment by chriskn** [comment]
Hi @ImFlog, thank you for the fast response. I will give it a try and open an PR in the next days.

**Comment by ImFlog** [comment]
Closed by #117 and available in version 1.10.0.
Thank you again for the contribution @chriskn 🙏
